### PR TITLE
Provide a stack trace on a lua error with flex backend

### LINKED ISSUE #1136: Provide a stack trace on a lua error with flex backend
If you have an error in your lua script you get an error message like
```
Osm2pgsql failed due to ERROR: Failed to execute lua processing function: openstreetmap-carto.lua:533: Error in 'add_row': Trying to add relation to table 'planet_osm_line'
```

This doesn't provide as much information as a stack trace.

### PR DESCRIPTION
**Pull request by mmd-osm**
Closes #1136

Almost exactly the same code that is in use in the Lua interpreter, except for the signal handling: https://github.com/lua/lua/blob/master/lua.c#L109-L142

### CODE REVIEW

**Comment by joto** [comment]
This code breaks about every convention we have in our source code.

**Comment by mmd-osm** [comment]
Ok, so what do you suggest moving forward?

File: src/output-flex.cpp [hunk 1]
```diff
@@ -983,6 +983,35 @@ void output_flex_t::add_row(table_connection_t *table_connection,
...
+    luaL_traceback(lua_state, lua_state, msg, 1); // append a standard traceback
+    return 1;                                     // return the traceback
+}
+
+int output_flex_t::do_pcall(int narg, int nres)
+{
+    int const base = lua_gettop(lua_state()) - narg; // function index
+    lua_pushcfunction(lua_state(), msghandler);      // push message handler
+    lua_insert(lua_state(), base);                   // put it under function and args
+    int status = lua_pcall(lua_state(), narg, nres, base);
```
Comment (joto) [inline src/output-flex.cpp:1010]: Can be `const`

File: src/output-flex.cpp [hunk 2]
```diff
@@ -983,6 +983,35 @@ void output_flex_t::add_row(table_connection_t *table_connection,
...
+{
+    assert(lua_state);
+
+    char const *msg = lua_tostring(lua_state, 1);
+    if (msg == nullptr) {                                // is error object not a string?
+        if (luaL_callmeta(lua_state, 1, "__tostring") && // does it have a metamethod
+            lua_type(lua_state, -1) == LUA_TSTRING) {    // that produces a string?
+            return 1;                                    // that is the message
+        }
+        else {
```
Comment (joto) [inline src/output-flex.cpp:996]: Formatting looks unusual. Did you run `clang-format`?

File: src/output-flex.cpp [hunk 3]
```diff
@@ -983,6 +983,35 @@ void output_flex_t::add_row(table_connection_t *table_connection,
     }
 }
 
+static int msghandler(lua_State *lua_state)
```
Comment (joto) [inline src/output-flex.cpp:986]: Maybe a bit more descriptive function name?

**Pull request merged**